db/c.cc

```diff
@@ -341,6 +351,174 @@ struct rocksdb_compactionfilter_t : public CompactionFilter {
   bool IgnoreSnapshots() const override { return ignore_snapshots_; }
 };
 
+/* Table Properties */
+
+struct rocksdb_table_properties_t {
+  const TableProperties* rep;
+};
+
+struct rocksdb_table_properties_collector_factory_context_t {
+  TablePropertiesCollectorFactory::Context* rep;
+};
+
+struct rocksdb_user_collected_properties_t {
+  UserCollectedProperties* rep;
+};
+
+struct rocksdb_table_properties_collector_t : public TablePropertiesCollector {
+  void* state_;
+  void (*destructor_)(void*);
+  bool (*add_user_key_)(void*, const char* key, size_t key_len,
```

Comment:
The `add_user_key_` and `finish_` callbacks currently return boolean, while the C++ counterparts in `TablePropertiesCollector` return `Status`. RocksDB doesn't do very much with the status other than check for OK/!OK - it logs if `AddUserKey` returns non-OK status but only the collector's name. I doesn't seem worthwhile extending support for handling arbitrary `Support` values.